php-classes/Slate/CBL/Tasks/TasksRequestHandler.php

```diff
@@ -163,11 +164,74 @@ public static function handleTaskAssigneesRequest(ActiveRecord $Record)
         ]);
     }
 
+
+    public static function applyRecordDelta(\ActiveRecord $Record, $requestData)
+    {
+        parent::applyRecordDelta($Record, $requestData);
+
+        $defaultAttachmentClass = AbstractTaskAttachment::$defaultClass;
+
+        if (isset($requestData['Attachments'])) {
+            $attachments = [];
+            foreach ($requestData['Attachments'] as $attachmentData) {
+                $attachmentClass = $attachmentData['Class'] ?: $defaultAttachmentClass;
+                if ($attachmentData['ID'] >= 1) {
+                    if (!$Attachment = $attachmentClass::getByID($attachmentData['ID'])) {
+                        $failed[] = $attachmentData;
+                        continue;
+                    }
+
+                    if (!empty($attachmentData['Status']) && in_array($attachmentData['Status'], $defaultAttachmentClass::getFieldOptions('Status', 'values'))) {
+                        $Attachment->Status = $attachmentData['Status'];
+                    }
+                } else {
+                    $Attachment = $attachmentClass::create($attachmentData);
+                }
+
+
+                if ($Attachment instanceof Attachments\GoogleDriveFile) {
+                    if (!$Attachment->File) {
+                        if (!$File = \Google\DriveFile::getByWhere(['DriveID' => $attachmentData['File']['DriveID']])) {
+                            $File = \Google\DriveFile::create($attachmentData['File']);
+                            if (!$File->OwnerEmail && $GLOBALS['Session']->Person && $GLOBALS['Session']->Person->PrimaryEmail) {
+                                $File->OwnerEmail = $GLOBALS['Session']->Person->PrimaryEmail->toString();
+                            }
+                        }
+                        $Attachment->File = $File;
+                    } else if ($Attachment->File->isPhantom && $File = \Google\DriveFile::getByWhere(['DriveID' => $attachmentData['File']['DriveID']])) {
+                        $Attachment->File = $File;
+                    }
+                }
+
+                $attachments[] = $Attachment;
+            }
+
+            $Record->Attachments = $attachments;
+        }
+        // update student tasks
+        if (isset($requestData['Assignees'])) {
+            $studentTasks = [];
+            foreach ($requestData['Assignees'] as $assigneeId) {
+                if (!$StudentTask = StudentTask::getByWhere(['StudentID' => $assigneeId, 'TaskID' => $Record->ID])) {
+                    $StudentTask = StudentTask::create([
+                        'StudentID' => $assigneeId,
+                        'SectionID' => $_GET['SectionID'],
```

Comment:
@nbey Why does this need to come in through the side as a GET param? If it's part of the record it should come in from `$requestData` with the record's other fields. If it's part of the saved record, it seems like the client code could just as easily declare it as a model field and set it before calling `record.save()` as easily as attaching custom request-level params to the save operation?